src/utils/i18n/i18nextLocize.ts

```diff
@@ -191,7 +192,83 @@ export const locizeBackendOptions = {
 };
 
 /**
- * Fetch translations from Locize API
+ * Builds Locize API endpoint based on the desired lang and namespace
+ *
+ * @param lang
+ * @param namespace
+ */
+export const buildLocizeAPIEndpoint = (lang: string, namespace: string = defaultNamespace): string => {
+  return locizeBackendOptions
+    .loadPath
+    .replace('{{projectId}}', locizeBackendOptions.projectId)
+    .replace('{{version}}', locizeBackendOptions.version)
+    .replace('{{lng}}', lang)
+    .replace('{{ns}}', namespace);
+};
+
+/**
+ * Fetches the translations that are shared amongst all customers.
+ *
+ * Used as base translations, which can be overridden using translations variation.
+ *
+ * @param lang
+ */
+export const fetchBaseTranslations = async (lang: string): Promise<I18nextResources> => {
+  const locizeAPIEndpoint: string = buildLocizeAPIEndpoint(lang);
+  let i18nTranslations: I18nextResources = {};
+
+  try {
+    // Manually fetching locales from Locize API, for the "common" namespace of the current language
+    // XXX We fetch manually from Locize, because if we use the i18next "preload" feature, it'll crash with Next (no serverless support)
+    logger.info(`Pre-fetching translations from ${locizeAPIEndpoint}`);
+    const defaultI18nTranslationsResponse: Response = await fetch(locizeAPIEndpoint);
+
+    try {
+      i18nTranslations = await defaultI18nTranslationsResponse.json();
+    } catch (e) {
+      logger.error(e.message, `Failed to extract JSON data from locize API response for "${lang}"`);
+      Sentry.captureException(e);
+    }
+  } catch (e) {
+    logger.error(e.message, `Failed to fetch data from locize API for "${lang}"`);
+    Sentry.captureException(e);
+  }
+
+  return i18nTranslations;
+};
+
+/**
+ * Fetches the translations that are specific to the customer (its own translations variation)
+ *
+ * @param lang
+ */
+export const fetchCustomerVariationTranslations = async (lang: string): Promise<I18nextResources> => {
```

Comment:
This is a personal opinion, but I don't like this variable naming convention of using `customer`, specially for users that are not using MST, which is my case, I don't have this concept of `customer`, it's just one single project.

Anyway, this is just my 2 cents.

Reply:
I understand, but this is specific to MST though, so it makes sense in this context.

I'm afraid removing "customer" term would confuse about what this actually does and how it's used